source/grid/grid_generator.cc

```diff
@@ -7250,29 +7363,101 @@ namespace GridGenerator
         // create cells one by one
 
         if (dim == 2)
-          for (const auto &cell : table_2D_cell)
-            add_cell(dim, cell, current_material_id);
+          {
+            const auto current_manifold_id_of_cell =
+              cell.manifold_id(); // get cell-manifold id from current quad cell
+            for (const auto &cell : table_2D_cell)
+              add_cell(dim,
+                       cell,
+                       current_material_id,
+                       current_manifold_id_of_cell); // set cell-manifold id
+
+            // inherit inner manifold (faces)
+            if (current_manifold_id_of_cell != numbers::flat_manifold_id)
+              for (const auto &face : table_2D_inner_faces)
```

Comment:
same comment about using a better name for the `face` variable. The same applies in the code pieces below.